lrauv_ignition_plugins/test/test_mission_pitch_mass.cc

```diff
@@ -76,21 +76,51 @@ TEST_F(LrauvTestFixture, PitchMass)
 
   ignmsg << "Logged [" << this->tethysPoses.size() << "] poses" << std::endl;
 
-  // Give it some iterations to reach steady state
-  for (auto i = 2000u; i < this->tethysPoses.size(); i = i + 1000)
+  int maxIterations{28000};
+  ASSERT_LT(maxIterations, this->tethysPoses.size());
+
+  double totalPitchChange = 0, prevPitch = 0;
+  bool firstPitch = false, reachedTarget = false;
+
+  // Vehicle should have a max pitch of 20 degrees
+  for(auto pose: this->tethysPoses)
   {
-    this->CheckRange(i,
-      // Target pose has no translation, roll or yaw, and 20 deg pitch
-      {0.0, 0.0, 0.0, 0.0, IGN_DTOR(20), 0},
-      // Y is kept pretty stable close to zero with low tolerances
-      // \TODO(chapulina) X and Y are meant to stay close to zero, but the
-      // vehicle goes forward (-X, +Z) slowly. That's caused by the pitch
-      // oscillation.
-      {4.7, 0.01, 1.8},
-      // Roll and yaw are kept pretty stable close to zero with low tolerances
-      // \TODO(chapulina) The pitch has a lot of oscillation, so we need the
-      // high tolerance
-      {IGN_DTOR(2), IGN_DTOR(21), IGN_DTOR(2)});
+    // Max Pitch 20 degrees. Allow 5 degrees error for oscillations.
+    // Min Pitch -5 degrees. Again allow 5 degrees error for oscillation.
+    EXPECT_LT(pose.Rot().Euler().Y(), IGN_DTOR(25));
+    EXPECT_GT(pose.Rot().Euler().Y(), IGN_DTOR(-5));
+
+    // No roll or yaw
+    EXPECT_NEAR(pose.Rot().Euler().X(), IGN_DTOR(0), 1e-3);
+    EXPECT_NEAR(pose.Rot().Euler().Z(), IGN_DTOR(0), 1e-3);
+
+    // Check position holds
+    EXPECT_NEAR(pose.Pos().X(), 0, 2e-1);
+    EXPECT_NEAR(pose.Pos().Y(), 0, 2e-1);
+    EXPECT_NEAR(pose.Pos().Z(), 0, 2e-1);
+
+    // Used later for oscillation check.
+    if (!firstPitch)
+    {
+      totalPitchChange += std::fabs(pose.Rot().Euler().Y() - prevPitch);
+    }
+
+    // Check if we cross the 20 degree mark
+    if (prevPitch <= IGN_DTOR(20) && pose.Rot().Euler().Y() >= IGN_DTOR(20))
+    {
+      reachedTarget = true;
+    }
+
+    prevPitch = pose.Rot().Euler().Y();
+    firstPitch = true;
   }
+
+  // Check for oscillation by summing over abs delta in pitch
+  // Essentially \Sigma abs(f'(x)) < C. In this case C should be near 2*20
+  // degrees as the vehicle first pitches down and then comes back up.
+  ASSERT_LE(totalPitchChange, IGN_DTOR(21) * 2);
+
+  // Make sure the vehicle actually pitched to 20 degrees.
+  ASSERT_TRUE(reachedTarget);
```

Comment:
I'm proposing we get this test into `main` before #89 and #96, see https://github.com/osrf/lrauv/pull/134.

Reply:
I don't agree with this approach for this test at all. It is coupled to this change. I can separate it and target this but the test should be an :x:  on the main. Tweaking tolerances is as good as commenting out the changes.

Reply:
Thanks for approving that PR. 

> Tweaking tolerances is as good as commenting out the changes.

I disagree. The tests serve the purpose of tracking the desired behaviour and how much we're deviating from it. Now that all branches are running the same test, the test tells a story:

1. On `main`, pitch mass is oscillating like crazy

    https://github.com/osrf/lrauv/blob/62c63a00e9e88fa6674fbf52c3151f1b0ffdf9ff/lrauv_ignition_plugins/test/test_mission_pitch_mass.cc#L88-L102

1. On #89, it oscillates even more, while other tests still pass. A reasonable question is what is #89 improving?

    https://github.com/osrf/lrauv/blob/d3c21f723a5a69bf25d1c228d98193dc32a0dee1/lrauv_ignition_plugins/test/test_mission_pitch_mass.cc#L88-L102

1. On this PR, the oscillation is largely gone, which is validated by the reduction in tolerance

    https://github.com/osrf/lrauv/blob/a2dc738630c0e425fb8b2e86e0ae511705a7dfc1/lrauv_ignition_plugins/test/test_mission_pitch_mass.cc#L88-L100

You must have gone through all these stages manually during implementation. But reviewers and future developers don't have the same experience. So it's helpful for others (well, at least me), to see these results.

Reply:
Thanks for doing this. It makes sense in hind sight.